applications/Unified_Liquidity_Balance_API.md

```diff
@@ -0,0 +1,158 @@
+
+### Overview
+
+- **Tagline:** Real‑time cross‑chain balance monitoring via SSE over HTTPS
+- **Description:** The Unified Liquidity Balance API is a real‑time balance tracking solution for Polkadot SDK-based chains. It provides a single, unified Server‑Sent Events (SSE) endpoint to monitor user balances across multiple parachains without polling or deep integration with Polkadot‑specific libraries.
+- **Integration:** Uses the Polkadot API to connect to Polkadot SDK-based chains (Polkadot, Asset Hub, Polimec, Hydration, etc.), deployed on Cloudflare Workers with Durable Objects to maintain state at the edge.
+- **Team Interest:** We aim to lower the barrier to building cross‑chain dApps by abstracting multi‑chain balance monitoring into a familiar web streaming interface, enabling web developers to adopt Polkadot technologies more easily.
+
+### Project Details
+
+- **Mockups / UI:** This is a backend service exposing SSE endpoints; no front‑end mockups required.
+- **Data Models / API Specification:**
+  - **Example Endpoint:** GET /balances/{accountId}
+  - **Event types:**
+    - `heartbeat`
+    - `accountId`
+    - `{tokenSymbol}` (e.g., `DOT`)
+  - **Balance Event Payload:**
+    ```json
+    {
+      "total": "string",
+      "chains": {
+        "Polkadot": "string",
+        "AssetHub": "string",
+        "Polimec": "string",
+        "Hydration": "string"
+      }
+    }
+    ```
+- **Technology Stack:**
+  - Cloudflare Workers (the Worker's code will be in TypeScript)
+  - Cloudflare Durable Objects
+  - [Polkadot-API](https://github.com/polkadot-api/polkadot-api) (PAPI) for chain connections
+  - RxJS for reactive streams
+- **PoC / MVP:**
+  - Deployed prototype: https://polkadot-cloud-sse.polimec.workers.dev/
+  - Repository: https://github.com/Polimec/ulb-api
+- **Core Component Documentation:**
+  1. **Chain Adapters** (`src/adapters/*`): connect to RPC endpoints, normalize balance data.
+  2. **Balance Service** (`src/services/balance.ts`): aggregates per‑chain data.
+  3. **Stream Service** (`src/services/stream.ts`): manages SSE connections and heartbeats.
+  4. **Durable Object Listener** (`src/durable/listener.ts`): persists subscriptions, orchestrates updates.
+- **Out of Scope:**
+  - Account/key management or transaction submission
+  - Support for non‑Polkadot SDK chains (initial version)
+
+### Ecosystem Fit
+
+- **Placement:** Infrastructure service for cross‑chain dApp/back‑end developers, complementary to RPC nodes and light clients.
+- **Target Audience:** dApp/web developers, wallet providers, and analytics platforms building on Polkadot who need real-time data without having to understand or use Polkadot-specific tools and libraries.
+- **Needs Met:** Unified, low‑latency, real-time balance monitoring across multiple parachains.
+- **Similar Projects:**
+  - Off‑chain indexing tools provide richer data but with higher setup overhead and they are not offering any SSE-based API.
```

Comment:
Could you elaborate on a specific scenario where your SSE-based API is distinctly advantageous over using something like SubQuery or Subquid indexer?